added moonplot branch

williyamshoe opened the pull request:
added moonplot to summary page

I don't know why there are so many indentation changes; I don't think I even touched those indents.

williyamshoe committed:
added moonplot branch

sbailey commented on py/surveyqa/summary.py:
```diff
@@ -395,6 +395,25 @@ def exptime_dgb(string, color):
     return fig
 
+def get_moonplot(exposures, width=500, height=300):
+    p = bk.figure(plot_width=width, plot_height=height, x_axis_label = "MOONFRAC", y_axis_label = "MOONALT")
```
Please add a docstring clarifying which columns of exposures are required to be able to make this plot.

sbailey commented on py/surveyqa/summary.py:
```diff
@@ -395,6 +395,25 @@ def exptime_dgb(string, color):
+
+    keep = exposures['PROGRAM'] != 'CALIB'
+    exposures_nocalib = exposures[keep]
+    color_mapper = LinearColorMapper(palette="Magma256", low=np.min(exposures_nocalib["MOONSEP"]), high=np.max(exposures_nocalib["MOONSEP"]))
```
Let's fix the low/high values to be 0-180 so that it has a consistent color mapping range regardless of the details of the input data.

sbailey commented on py/surveyqa/summary.py:
```diff
@@ -435,36 +454,36 @@ def makeplots(exposures, tiles, outdir):
         margin: 0;
     }
-    
+
```
All these whitespace changes are likely due to your editor being configured to auto-strip unnecessary trailing whitespace.  If you can find a way to turn that off, please do.  No need to try to undo these edits here though -- in general it is "right" to not have unnecessary trailing whitespace, but it does risk merge conflicts when your editor automatically makes updates you didn't intend.

sbailey commented on py/surveyqa/summary.py:
```diff
@@ -482,38 +501,39 @@ def makeplots(exposures, tiles, outdir):
-                
+                    <div>{{ moonplot_script }} {{ moonplot_div }}</div>
+                </div>
+
```
This moonplot got added as part of the "progress graphs" section, which seems misplaced.  It fits more with the histograms, while renaming that section to something like "Observing Conditions".

@ana-lyons please confirm that this would be consistent with any updates that you are working on to avoid merge conflicts.

ana-lyons replied:
as it is right now, it looks fine with what I have. should I move the moonplot graph to the histograms section and rename it? or should William do that.

sbailey replied:
Thanks for the confirmation @ana-lyons.  Let's move the div and rename the section within this branch before merging.  @williyamshoe can do that as part of the other requested updates in this PR.

williyamshoe replied:
Alright, will do. I wasn't sure where to put the plot since I felt it didn't fit in the progress nor the histogram section. I'll move it to the histogram section and rename the section to "Observing Conditions", along with committing the other changes listed above.

sbailey commented:
@williyamshoe please post a comment in this PR when you are done making updates and have pushed them to github, i.e. when this is ready for re-review.  Thanks.

williyamshoe committed:
adjusted docstrings, moonplot color distribution, and location of plot

williyamshoe commented:
Alright; pushed my changes to the branch. I added docstrings to all of my functions, changed the color distribution to a fixed [0, 180], moved the moonplot down to the histograms section, and renamed the histograms section to "Observing Conditions".

sbailey commented:
Looks good, thanks.  Merging now.
